src/getchar.c

```diff
@@ -1095,6 +1141,26 @@ ins_typebuf(
     return OK;
 }
 
+/*
+Safely adds two integers
+
+Returns -1 if the result of a+b results is an overflow
+Returns the sum otherwise
```

Comment:
The comment doesn't match the behaviour.

Reply:
You are completely right. What a catch!